src/Worker/UpdateContact.php

```diff
@@ -22,20 +22,39 @@
 namespace Friendica\Worker;
 
 use Friendica\Core\Logger;
+use Friendica\Core\Worker;
 use Friendica\Model\Contact;
+use Friendica\Network\HTTPException\InternalServerErrorException;
 
 class UpdateContact
 {
 	/**
 	 * Update contact data via probe
 	 *
-	 * @param int    $contact_id Contact ID
+	 * @param int $contact_id Contact ID
 	 * @return void
+	 * @throws InternalServerErrorException
+	 * @throws \ImagickException
 	 */
 	public static function execute(int $contact_id)
 	{
 		$success = Contact::updateFromProbe($contact_id);
 
 		Logger::info('Updated from probe', ['id' => $contact_id, 'success' => $success]);
 	}
+
+	/**
+	 * @param array|int $run_parameters Priority constant or array of options described in Worker::add
+	 * @param int       $contact_id
+	 * @return int
+	 * @throws InternalServerErrorException
+	 */
+	public static function add($run_parameters, int $contact_id): int
```

Comment:
Aren't the files in `/Worker/` expected to only be called via a worker, thus only containing the `execute` function?

Reply:
They need to have at least one `execute()` method, something we could enforce with an interface, but they can also have additional methods.

Reply:
Of course they *can* - but *should* they? I think that it is the same like executing functionality in a `Module` class from something else than the router.

Reply:
I'm with @annando , we should use a class in its namespaced scope, and Worker classes should be used for Worker purposes :-)

Reply:
We’ve been having these issues where worker tasks are being created with wrong parameters. Unfortunately we can’t track this with `Worker:add` because of the arbitrary parameter array, so I thought that each worker class could validate its own parameters in a similar `add()` method.

I don’t think this method is outside of the worker scope and it allows for more fine-grained parameter validation, including figuring out where the parameter was wrongly supplied, something we cannot physically do at the moment.

Reply:
I understand what you mean. But shouldn't the function name at least mention "worker" in it? Like `AddToWorker`?

Reply:
“Worker” is already present in the class namespace, I don’t think it needs to be repeated.

Reply:
Since this name will most likely be used with other worker classes as well, I delegate this to @nupplaphil :grimacing:

Reply:
@MrPetovan 

OK, another try:

What about using some kind of callback function when using `Worker::add()` like we do for Hooks to check parameters, if there's a method?

like `UpdateContact::checkParameter(array $parameter)` and called like
```php
if (method_exists($className, 'checkParameter')) {
     call_user_func($className .'::checkParameter');
}
```

This will decouple the context and the name(space) pattern :)

Reply:
I still don't understand the problem with the namespace? My solution looks more straightforward to me than what you're suggesting. I don't like using either `method_exists` or `call_user_function` because IDEs can't follow the call chain.

Do I need to write an interface that all worker tasks should be implementing with both the `execute` and the `add` methods?

Reply:
hmm .. maybe I'm wrong with my thinking about the context .. I don't know ..

Since it's "just" as design thing and we need the bugfix, I'll go on and don't stand in the path anymore :D